freva-rest/src/databrowser_api/core.py

```diff
@@ -1115,19 +1170,33 @@ async def _ingest_user_metadata(self,
         -------
         None
         """
+        processed_metadata = [
+            {**metadata, **self.fwrites}
+            for metadata in user_metadata
+        ]
+        for i in range(0, len(processed_metadata), self.batch_size):
+            batch = processed_metadata[i:i + self.batch_size]
+            processed_batch = await self._process_metadata(batch)
+            if processed_batch:
+                await self._add_to_solr(processed_batch)
+                await self._insert_to_mongo(processed_batch)
+
+    async def _process_metadata(
+        self,
+        metadata_batch: List[Dict[str, Union[str, List[str], Dict[str, str]]]]
+    ) -> List[Dict[str, Union[str, List[str], Dict[str, str]]]]:
+        """Process the metadata batch by removing duplicates before ingestion."""
+        new_querie = []
+        for metadata in metadata_batch:
+            uri = metadata.get('uri', '')
+            file_path = metadata.get('file', '')
 
-        user_metadata_batch = []
-        for metadata in user_metadata:
-            metadata.update(self.fwrites)
-            user_metadata_batch.append(metadata)
-            if len(user_metadata_batch) >= self.batch_size:
-                await self._add_to_solr(user_metadata_batch)
-                await self._insert_to_mongo(user_metadata_batch)
-                user_metadata_batch = []
-
-        if user_metadata_batch:
-            await self._add_to_solr(user_metadata_batch)
-            await self._insert_to_mongo(user_metadata_batch)
+            if not uri and not file_path:
+                continue
+            is_duplicate = await self._is_query_duplicate(str(uri), str(file_path))
```

Comment:
You could also collect the number of duplicates and then add this to the message that n entries were not added because they were duplicates.